lib/puppet/provider/package/apt.rb

```diff
@@ -147,14 +147,32 @@ def install
     end
 
     cmd += install_options if @resource[:install_options]
-    cmd << :install << str
+    cmd << :install
+
+    if source
+      cmd << source
+    else
+      cmd << str
+    end
 
     self.unhold if self.properties[:mark] == :hold
     begin
       aptget(*cmd)
     ensure
       self.hold if @resource[:mark] == :hold
     end
+
+    # If a source file was specified, we must make sure the expected version was installed from specified file
+    if source && !should.is_a?(Symbol)
```

Comment:
why doe we need `!should.is_a?(Symbol)`?

Reply:
if the install command was successful, the package should have been installed, extra-check is not needed in case if `:present` or `:installed` values
we do extra-checks only in case specific version was requested

Reply:
Rather than relying on type, maybe we should check the actual values.